src/Raven.Client/Documents/Operations/Attachments/GetAttachmentsOperation.cs

```diff
@@ -0,0 +1,150 @@
+                            writer.WriteComma();
+                            writer.WritePropertyName("ReadAll");
+                            writer.WriteBool(_readAll);
+
+                            if (_readAll == false)
+                            {
+                                writer.WriteComma();
+                                writer.WriteArray("Names", _names);
+                            }
+
+                            writer.WriteEndObject();
+                        }
+                    })
+                };
+
+                return request;
+            }
+
+            public override async Task<ResponseDisposeHandling> ProcessResponse(JsonOperationContext context, HttpCache cache, HttpResponseMessage response, string url)
+            {
+                context.Reset();
+                context.Renew();
+
+                AttachmentsDetails attachmentsMetadata;
+                AttachmentsStreamInfo streamInfo;
+                var state = new JsonParserState();
+                Stream stream = response.Content.ReadAsStreamAsync().Result;
+                context.GetManagedBuffer(out JsonOperationContext.ManagedPinnedBuffer buffer);
+
+                using (var parser = new UnmanagedJsonParser(context, state, "attachments/receive"))
+                using (var builder = new BlittableJsonDocumentBuilder(context, BlittableJsonDocumentBuilder.UsageMode.None, "attachments/list", parser, state))
+                using (var peepingTomStream = new PeepingTomStream(stream, context))
+                {
+                    if (await UnmanagedJsonParserHelper.ReadAsync(peepingTomStream, parser, state, buffer).ConfigureAwait(false) == false)
+                        throw new Exception("cannot parse stream");
+
+                    if (state.CurrentTokenType != JsonParserToken.StartObject)
+                        throw new Exception($"Expected token {nameof(JsonParserToken.StartObject)}, but got {nameof(state.CurrentTokenType)}.");
+
+                    await UnmanagedJsonParserHelper.ReadObjectAsync(builder, peepingTomStream, parser, buffer).ConfigureAwait(false);
+
+                    BlittableJsonReaderObject data = builder.CreateReader();
+                    attachmentsMetadata = JsonDeserializationClient.AttachmentAdvancedDetails(data);
+
+                    buffer.Used = parser.BufferOffset;
+                    buffer.Valid = parser.BufferSize;
+
+                    streamInfo = new AttachmentsStreamInfo
+                    {
+                        AttachmentAdvancedDetails = attachmentsMetadata,
+                        Buffer = buffer
+                    };
+                }
+
+                Result = attachmentsMetadata.AttachmentsMetadata.ToDictionary(attachment => attachment.Name, attachment => new AttachmentResult
+                {
+                    Stream = new AttachmentsStream(stream, attachment.Name, attachment.Index, (int)attachment.Size, streamInfo),
```

Comment:
Why cast int? We can have documents larger than 2GB

Reply:
done